messaging/src/test/java/org/axonframework/tracing/IntermediateSpanFactoryTest.java

```diff
@@ -0,0 +1,97 @@
+package org.axonframework.tracing;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.TestDescriptor;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Abstract class test for all intermediate factories that result in a {@link Span}. Contains facilities
+ * to setup tests in a good manner and run them.
+ *
+ * @param <BI> The builder class
+ * @param <SI> The span factory class
+ * @since 4.9.0
+ * @author Mitchell Herrijgers
+ */
+public abstract class IntermediateSpanFactoryTest<BI, SI> {
+    private final SpanFactory spanFactory = new TestSpanFactory();
+
+    /**
+     * Creates a new factory builder that also sets the provided span factory.
+     */
+    protected abstract BI createBuilder(SpanFactory spanFactory);
+
+    /**
+     * Executes the builder method and returns the factory based on the builder.
+     */
+    protected abstract SI createFactoryBasedOnBuilder(BI builder);
+
+    @Test
+    void cannotSetNullSpanFactory() {
+        Assertions.assertThrows(AxonConfigurationException.class, () -> createBuilder(null));
+    }
+
+    protected void test(Function<BI, BI> builderAdditions, Function<SI, Span> invocation, SpanDefinition definition) {
```

Comment:
Although `protected` methods strictly speaking don't need Javadoc, the fact this will be used throughout for the numerous "intermediate" `SpanFactory` instances you're creating, I think it's valuable to add a little text here too.

Reply:
Added